src/network.rs

```diff
@@ -54,25 +53,57 @@ impl Network {
     }
 
     /// Download a file and persist it to disk
-    pub async fn download_and_persist(&self, url: Url, persist_to: &Path) -> Result<(), VetError> {
-        let download_tmp_path = OsString::from_iter([persist_to.as_os_str(), OsStr::new(".part")]);
+    pub async fn download_and_persist(
```

Comment:
I feel like with some functions like this what we really want is to apply some wrapper to the error as it leaves this function, but we really don't need this much granularity for each piece of error in the function.